BinaryEncoding.md

```diff
@@ -240,6 +239,42 @@ This section must be preceded by a [Functions](#functions-section) section.
 | count | `varuint32` | count of entries to follow |
 | entries | `uint16*` | repeated indexes into the function table |
 
+### Function Names section
+
+ID: `function_names`
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| names | `utf8-C-string*` | sequence of null-terminated utf8-encoded strings |
```

Comment:
To avoid the annoying (and sure to cause bugs) question of "what happens if null is in the middle of the string", I'd like to disallow that case regardless.  So if we wanted up-front length, then we'd also drop the null-terminator (otherwise it's just more work to make sure both expressions of length match) right?  That'd be fine with me.